src/Form/LinkBlockFormDataProvider.php

```diff
@@ -0,0 +1,283 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\LinkList\Form;
+
+use PrestaShop\Module\LinkList\Cache\LinkBlockCacheInterface;
+use PrestaShop\Module\LinkList\Model\LinkBlock;
+use PrestaShop\Module\LinkList\Repository\LinkBlockRepository;
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleRepository;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use Hook;
+
+/**
+ * Class LinkBlockFormDataProvider.
+ */
+class LinkBlockFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var int|null
+     */
+    private $idLinkBlock;
+
+    /**
+     * @var LinkBlockRepository
+     */
+    private $repository;
+
+    /**
+     * @var LinkBlockCacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var ModuleRepository
+     */
+    private $moduleRepository;
+
+    /**
+     * @var array
+     */
+    private $languages;
+
+    /**
+     * @var int
+     */
+    private $shopId;
+
+    /**
+     * LinkBlockFormDataProvider constructor.
+     *
+     * @param LinkBlockRepository $repository
+     * @param LinkBlockCacheInterface $cache
+     * @param ModuleRepository $moduleRepository
+     * @param array $languages
+     * @param int $shopId
+     */
+    public function __construct(
+        LinkBlockRepository $repository,
+        LinkBlockCacheInterface $cache,
+        ModuleRepository $moduleRepository,
+        array $languages,
+        $shopId
+    ) {
+        $this->repository = $repository;
+        $this->cache = $cache;
+        $this->moduleRepository = $moduleRepository;
+        $this->languages = $languages;
+        $this->shopId = $shopId;
+    }
+
+    /**
+     * @return array
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function getData()
+    {
+        if (null === $this->idLinkBlock) {
+            return [];
+        }
+
+        $linkBlock = new LinkBlock($this->idLinkBlock);
+
+        $arrayLinkBlock = (array) $linkBlock;
```

Comment:
why not adding a `->toArray` function?

Reply:
I think it is the responsibility of the FormDataProvider to represent the form data the correct way, the LinkBlock class has no knowledge of how the form expects its data

Reply:
oh I misunderstood what you meant, ok for the toArray method then you're right but the transformation of the data will still be in the FormDataProvider